src/engine/sparqlExpressions/ConvertToDtypeConstructor.cpp

```diff
@@ -103,8 +124,55 @@ class ToBooleanImpl {
 using ToBoolean = NARY<1, FV<ToBooleanImpl, ToNumericValueGetter>>;
 }  // namespace detail::to_boolean
 
+//______________________________________________________________________________
+// CONVERT TO DATE(TIME)
+namespace detail::to_datetime {
+
+// Cast to xsd:dateTime or xsd:date (ValueId)
+template <bool ToJustXsdDate>
+inline auto convertStringToDateTimeValueId =
+    [](OptIdOrString input) -> ValueId {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  }
+  const auto& inputValue = input.value();
+
+  // Remark: If the parsing procedure for datetime/date string values with
+  // parseXsdDatetimeGetOptDate/parseXsdDateGetOptDate fails,
+  // return Id::makeUndefined().
+  const auto retrieveValueId = [](std::optional<DateYearOrDuration> optValue) {
+    if (optValue.has_value()) {
+      return Id::makeFromDate(optValue.value());
+    }
+    return Id::makeUndefined();
+  };
```

Comment:
If I am reading this correctly, then
`xsd:date(someXsdDateTimeId)` will still return the the `xsdDateTime` and not cast it to a plain date.
I have no strong opinion on this, as it is not really important for the SPARQL standard support, but we could
discuss whether we want to cast ` 2025-01-01 13:10:12^^dateTime` to `2025-01-01^^date` in this case
@hannahbast Do you have an opinion on this?